src/AnchoredOverlay/AnchoredOverlay.tsx

```diff
@@ -0,0 +1,126 @@
+import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react'
+import Overlay from '../Overlay'
+import {useFocusTrap} from '../hooks/useFocusTrap'
+import {useFocusZone} from '../hooks/useFocusZone'
+import {useAnchoredPosition, useRenderForcingRef} from '../hooks'
+import {uniqueId} from '../utils/uniqueId'
+
+export interface AnchoredOverlayProps {
+  /**
+   * A custom function component used to render the anchor element.
+   * Will receive the selected text as `children` prop when an item is activated.
+   */
+  renderAnchor: <T extends React.HTMLAttributes<HTMLElement>>(props: T) => JSX.Element
+
+  /**
+   * Determines whether the overlay portion of the component should be shown or not
+   */
+  open: boolean
+
+  /**
+   * A callback which is called whenever the overlay is currently closed and an "open gesture" is detected.
+   */
+  onOpen?: (gesture: 'anchor-click' | 'anchor-key-press') => unknown
+
+  /**
+   * A callback which is called whenever the overlay is currently open and a "close gesture" is detected.
+   */
+  onClose?: (gesture: 'click-outside' | 'escape') => unknown
+}
+
+/**
+ * An `AnchoredOverlay` provides an anchor (button by default) that will open a floating overlay.
+ * The overlay can be opened and navigated using keyboard or mouse.
+ */
+export const AnchoredOverlay: React.FC<AnchoredOverlayProps> = ({renderAnchor, children, open, onOpen, onClose}) => {
+  const anchorRef = useRef<HTMLButtonElement>(null)
```

Comment:
Should we make this more general to cover cases when the anchor is not a button? Or do we want to enforce that the anchor is always a button?
```suggestion
  const anchorRef = useRef<HTMLElement>(null)
```

Reply:
Good suggestion!  I think there was something weird with the default being a button that cause me to change this, but `HTMLElement` totally works now 😄